scenes/player/weapon/Weapon.gd

```diff
@@ -0,0 +1,55 @@
+extends Node2D
+
+export var hit_umbral = 0.25
+var elpased_hit_time = 0
+var is_flipped = false
+
+signal hit
+signal hit_attempt_started
+signal hit_attempt_ended
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	pass # Replace with function body.
+
+
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+func _process(delta):
+	self.elpased_hit_time += delta
+	
+	if self.elpased_hit_time > hit_umbral and Input.is_action_just_pressed("ui_attack"):
+		self.behave_when_hitting()
+
+func behave_when_hitting():
+	self.elpased_hit_time = 0
+	var rotation_factor = 1
+	if self.is_flipped:
+		rotation_factor = -1
+	emit_signal("hit_attempt_started")
+	$Pivot.rotate(0.6 * PI * rotation_factor)
+		
+	var timer = Timer.new()
```

Comment:
Can we do this by adding a Timer node to the scene via the godot editor instead?